src/test/java/jcifs/tests/SymLinkTest.java

```diff
@@ -0,0 +1,188 @@
+        assertNotNull(smbFile);
+
+        SmbFile[] files = smbFile.listFiles();
+        log.info(Arrays.toString(files));
+        assertNotNull(files);
+        assertTrue("No share found", files.length > 0);
+
+        for (SmbFile file : files) {
+            assertNotNull(file);
+
+            if (file.isDirectory()) {
+                log.info("Resource [" + file.getName() + "] is a directory.");
+            }
+            else {
+                log.info("Resource [" + file.getName() + "] is not a directory.");
+            }
+
+            if (file.isFile()) {
+                log.info("Resource [" + file.getName() + "] is a file.");
+            }
+            else {
+                log.info("Resource [" + file.getName() + "] is not a file.");
+            }
+
+            if (file.isSymlink()) {
+                log.info("Resource [" + file.getName() + "] is a symbolic link.");
+            }
+            else {
+                log.info("Resource [" + file.getName() + "] is not a symbolic link.");
+            }
+
+            try {
+                InputStream is = file.getInputStream();
+                assertNotNull(is);
+
+                if (file.isDirectory()) {
+                    log.info("file.list() -> {}", Arrays.toString(file.list()));
+
+                    if (file.getName().equals("symlinkTestDir/")) {
+                        SmbFile[] files2 = file.listFiles();
+                        assertNotNull(files2);
+                        log.info("file.listFiles() -> {}", Arrays.toString(files2));
+
+                        for (SmbFile file2 : files2) {
+                            InputStream is2 = file2.getInputStream();
+                            assertNotNull(is2);
+
+                            long l2 = copyInputStreamToFile(is2, File.createTempFile(file2.getName(), ".txt"));
+                            assertTrue("No bytes written", l2 > 0);
+                            assertTrue("Bytes written should be 10", l2 == 10);
+                        }
+                    }
+                }
+
+                if (file.isFile() && file.getName().equals("symlinkTestFoo.file")) {
+                    long l = copyInputStreamToFile(is, File.createTempFile(file.getName(), ".txt"));
+                    assertTrue("No bytes written", l > 0);
+                    assertTrue("Bytes written should be 9", l == 9);
+                }
+            }
+            catch (IOException ioe) {
+                throw new SmbException("testSymlink1 error", ioe);
+            }
+            finally {
+                file.close();
+            }
+        }
+
+        smbFile.close();
+    }
+
+
+    private SmbFile createSession() throws MalformedURLException, CIFSException {
+        Properties props = new Properties();
+        props.setProperty("jcifs.smb.client.minVersion", DialectVersion.SMB210.name());
+        props.setProperty("jcifs.smb.client.maxVersion", DialectVersion.SMB311.name());
```

Comment:
The way I use the test configurations it would be better to have a version check as precondition (Assume) instead. And the minimum version should be 3.1.1.
Then the default context/utility functions can be used as well.

Reply:
Same comment as above...